components/infobox/commons/infobox_widget_header.lua

```diff
@@ -92,49 +95,40 @@ end
 ---@param size number|string|nil
 ---@return Html?
 function Header:_image(fileName, fileNameDark, default, defaultDark, size)
-	if (fileName == nil or fileName == '') and (default == nil or default == '') then
+	local imageName = fileName or default
+	local imageDarkname = fileNameDark or fileName or defaultDark or default
+
+	if String.isEmpty(imageName) and String.isEmpty(imageDarkname) then
 		return nil
 	end
 
-	local imageName = fileName or default
-	---@cast imageName -nil
-	local infoboxImage = Header:_makeSizedImage(imageName, fileName, size, 'lightmode')
+	if imageName == imageDarkname then
+		return mw.html.create('div'):node(Header:_makeSizedImage(imageName, size))
+	end
 
-	imageName = fileNameDark or fileName or defaultDark or default
-	---@cast imageName -nil
-	local infoboxImageDark = Header:_makeSizedImage(imageName, fileNameDark or fileName, size, 'darkmode')
+	local infoboxImage = Header:_makeSizedImage(imageName, size, 'lightmode')
+	local infoboxImageDark = Header:_makeSizedImage(imageDarkname, size, 'darkmode')
 
 	return mw.html.create('div'):node(infoboxImage):node(infoboxImageDark)
 end
 
----@param imageName string
----@param fileName string?
+---@param imageName string?
 ---@param size number|string|nil
----@param mode string
----@return Html
-function Header:_makeSizedImage(imageName, fileName, size, mode)
-	local infoboxImage = mw.html.create('div'):addClass('infobox-image ' .. mode)
+---@param mode string?
+---@return Html?
+function Header:_makeSizedImage(imageName, size, mode)
+	if String.isEmpty(imageName) then
+		return
+	end
 
-	-- Number (interpret as pixels)
-	size = size or ''
-	if tonumber(size) then
-		size = tonumber(size) .. 'px'
+	local infoboxImage = mw.html.create('div'):addClass('infobox-image'):addClass(mode)
```

Comment:
Could the css be tweaked so `infobox-image` is only needed to be set on the wrapper?

Reply:
```
<div class="...-infobox">
 ...
 <div>
  <div class="infobox-image">
   <div class="lightmode">
     ...
   </div>
  </div>
 </div>
</div>
```
How about this

Reply:
do we even need the lightmode set as class in the div?
if using module:Image it would get set in the image itself

```lua
<div class="...-infobox">
 ...
 <div>
  <div class="infobox-image [infobox-fixed-size-image]">
   [File:...|class=show-when-light-mode][File:...|class=show-when-dark-mode] <-- Module Image call
  </div>
 </div>
 ...
</div>
```

would be in line with how other infobox rows are displayed too

![Screenshot 2024-04-22 074735](https://github.com/Liquipedia/Lua-Modules/assets/75081997/8f51b7a6-893e-4878-9984-4290f48c00c1)
(obviously in this case it would merke the 2 images into 1 since they are the same, just for showcasing with 2 different light/dark mode)